### DISCUSSION TITLE
Add declaration for dervied values in templates

### DISCUSSION THREAD

**Discussion by Sdju** [vue-macros/vue-macros#845] [Ideas]
Use cases:
- Processing values inside scoped slots
- Processing values inside v-for (without creating a computed property for values)

Pros:
- Simple
- Handy

Cons:
- More logic inside templates

Current solutions and their limitations:
1. Create a computed property (impossible for values inside scoped slots)
2. Create methods for processing values. unoptimized (e.g., when processing the same value multiple times)
3. Use :set / :data-set hack - it's a hack and doesn't work for Vapor (currently)
4. Create a simple utility component - while this is currently the best option, it requires recreating the component each time

Given these limitations, I think we need a way to create derived values inside templates and Vue Macros seems like the perfect place to test this feature.

Proposed syntax:
We already have a similar syntax for scoped params, so I suggest something similar:

```vue
<div  v-let="{ a = msg + '!', b = msg + '?' }">
  {{ a }} {{ b }}
</div>

// OR

<div  v-const="{ a: msg + '!', b: msg + '?' }">
  {{ a }} {{ b }}
</div>
```

Thank you

**Comment by sxzz** [comment]
1. It seems `{ a = msg + '!', b = msg + '?' }` is not a valid JS expression.
2. Are there any differences with the [v-scope RFC](https://github.com/vuejs/rfcs/discussions/505)?
